code/src/components/Task.jsx

```diff
@@ -0,0 +1,261 @@
+        taskindex: taskIndex,
+        projectindex: projectindex,
+      })
+    );
+  };
+
+  const deleteTask = () => {
+    dispatch(
+      projects.actions.deleteTask({
+        taskindex: taskIndex,
+        projectindex: projectindex,
+      })
+    );
+  };
+
+  const saveNewText = () => {
+    dispatch(
+      projects.actions.editTask({
+        taskindex: taskIndex,
+        projectindex: projectindex,
+        title: newText,
+      })
+    );
+  };
+
+  const editTask = () => {
+    setEditable(true);
+  };
+
+  const changingText = (e) => {
+    setNewText(e.currentTarget.textContent);
+  };
+
+  const pasteAsPlainText = (event) => {
+    event.preventDefault();
+    const text = event.clipboardData.getData("text/plain");
+    document.execCommand("insertHTML", false, text);
+  };
+
+  const disableNewLines = (e) => {
+    e.preventDefault();
+  };
+
+  const savingText = () => {
+    setEditable(false);
+    saveNewText();
+  };
+
+  return (
+    <TaskWrapper>
+      <CustomCheck
+        role="button"
+        onClick={toggleComplete}
+        complete={thisTask.complete}
+        tabIndex="0"
+        onKeyDown={(e) => e.key === "Enter" && toggleComplete()}
+        aria-label={
+          thisTask.complete
+            ? `task complete. press enter to toggle.`
+            : `task incomplete. press enter to toggle.`
+        }
+      >
+        {thisTask.complete ? "" : <IncompleteCheckIcon src={checkblack} />}
+        {thisTask.complete ? <CheckIcon src={checkblack} /> : ""}
+      </CustomCheck>
+      <TaskInfo complete={thisTask.complete}>
+        <TaskTitle
+          contentEditable={editable}
+          onBlur={(e) => changingText(e)}
+          onPaste={pasteAsPlainText}
+          onKeyPress={(e) => e.key === "Enter" && disableNewLines(e)}
+          tabIndex="0"
+        >
+          {thisTask.title}
+        </TaskTitle>
+        <TaskDate tabIndex="0">Added {created}</TaskDate>
+        <Line />
+      </TaskInfo>
+      <IconWrapper>
+        {!editable && (
+          <IconButton
+            role="button"
+            tabIndex="0"
+            onClick={editTask}
+            onKeyDown={(e) => e.key === "Enter" && editTask()}
+          >
+            <Icon src={editblack} alt="edit task." />
+          </IconButton>
+        )}
+        {editable && (
+          <IconButton
+            role="button"
+            tabIndex="0"
+            onClick={savingText}
+            onKeyDown={(e) => e.key === "Enter" && savingText()}
+          >
+            <Icon src={saveblack} alt="save edit." />
+          </IconButton>
+        )}
+
+        <IconButton
+          role="button"
+          tabIndex="0"
+          onClick={deleteTask}
+          onKeyDown={(e) => e.key === "Enter" && deleteTask()}
+        >
+          <Icon src={trashgray} alt="delete task." />
+        </IconButton>
+      </IconWrapper>
+    </TaskWrapper>
+  );
+};
+
+export default Task;
+
+const TaskWrapper = styled.div`
```

Comment:
Good idea to put the styling after all the logic since this component is very long! Functionality comes first ;)